js/dispatch_json.ts

```diff
@@ -30,11 +30,10 @@ function nextPromiseId(): number {
 }
 
 function decode(ui8: Uint8Array): JsonResponse {
+  // Received JSON string may be padded with "space" characters
+  // at the end, so make sure to trim it before parsing.
   let s = new TextDecoder().decode(ui8);
-  // TODO: this is make-shift solution
-  const closingBracket = s.lastIndexOf("}");
-  s = s.slice(0, closingBracket + 1);
-  return JSON.parse(s) as JsonResponse;
+  return JSON.parse(s.trimEnd()) as JsonResponse;
```

Comment:
Is there any need to trim it?

Reply:
Yes, take a look at above comment

Reply:
`JSON.parse()` will just ignore whitespace, won't it?

Reply:
@nayeemrmn good catch, it will. Removed the trim, thanks :+1: